### PR TITLE
BAEL-3583: JAAS

### PR DESCRIPTION
**Pull request by eelhazati**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -636,6 +636,7 @@
                 <module>webrtc</module>
                 <module>wildfly</module>
                 <module>quarkus-extension</module>
+				<module>security-jaas</module>
```
Comment (eric-martin) [inline pom.xml]: tabs/spaces

File: security-jaas/jaas-login-module/src/main/java/com/baeldung/security/jaas/authentication/InMemoryLoginModule.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+    private String username;
+    private boolean loginSucceeded = false;
+    private Principal userPrincipal;
+
+    @Override
+    public void initialize(Subject subject,
+                           CallbackHandler callbackHandler,
+                           Map<String, ?> sharedState,
+                           Map<String, ?> options) {
```
Comment (eric-martin) [inline security-jaas/jaas-login-module/src/main/java/com/baeldung/security/jaas/authentication/InMemoryLoginModule.java]: Follow our standard multi-line continuation rules here

File: security-jaas/jaas.policy [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+   permission java.lang.RuntimePermission "readFileDescriptor";
+   permission java.lang.RuntimePermission "writeFileDescriptor";
+};
+
+grant codebase "file:./jaas-login-module/target/jaas-login-module.jar" {
+    permission javax.security.auth.AuthPermission "modifyPrincipals";
+};
+
+grant principal com.sun.security.auth.UserPrincipal "testuser" {
+      permission com.baeldung.security.jaas.app.ResourcePermission "test_resource";
```
Comment (eric-martin) [inline security-jaas/jaas.policy]: 4 space indent

File: security-jaas/jaas-login-module/src/main/java/com/baeldung/security/jaas/authentication/InMemoryLoginModule.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    private static final String PASSWORD = "testpassword";
+
+    private Subject subject;
+    private CallbackHandler callbackHandler;
+    private Map<String, ?> sharedState;
+    private Map<String, ?> options;
+
+    private String username;
+    private boolean loginSucceeded = false;
+    private Principal userPrincipal;
```
Comment (eric-martin) [inline security-jaas/jaas-login-module/src/main/java/com/baeldung/security/jaas/authentication/InMemoryLoginModule.java]: Is it safe to save state in the module with private variables? Does each login get a new instance of the Module?

Comment (eelhazati) [reply security-jaas/jaas-login-module/src/main/java/com/baeldung/security/jaas/authentication/InMemoryLoginModule.java]: Yes each LoginContext#Login() ==> Call of all LoginModule#login() in a first pass and call of LoginModule#commit() in a second pass.

Here is the LoginContext#login() method:
    public void login() throws LoginException {

        loginSucceeded = false;

        if (subject == null) {
            subject = new Subject();
        }

        try {
            // module invoked in doPrivileged
            invokePriv(LOGIN_METHOD);
            invokePriv(COMMIT_METHOD);
            loginSucceeded = true;
        } catch (LoginException le) {

File: security-jaas/jaas-app/src/main/java/com/baeldung/security/jaas/app/JaasAuthentication.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+import javax.security.auth.Subject;
+import javax.security.auth.login.LoginException;
+
+public class JaasAuthentication {
+
+    public static void main(String[] args) throws LoginException {
+        LoginService loginService = new LoginService();
+        Subject subject = loginService.login();
+        System.out.println(subject.getPrincipals().iterator().next() + " sucessfully logeed in");
```
Comment (KevinGilmore) [inline security-jaas/jaas-app/src/main/java/com/baeldung/security/jaas/app/JaasAuthentication.java]:
```suggestion
        System.out.println(subject.getPrincipals().iterator().next() + " sucessfully logged in");
```

File: pom.xml [hunk 6]
```diff
@@ -431,6 +431,7 @@
                 <module>core-java-modules/core-java-nio</module>
                 <module>core-java-modules/core-java-nio-2</module>
                 <module>core-java-modules/core-java-security</module>
+				<module>core-java-modules/core-java-security-2</module>
```
Comment (eric-martin) [inline pom.xml]: Tabs/spaces

File: pom.xml [hunk 7]
```diff
@@ -1072,6 +1073,7 @@
...
```
Comment (eric-martin) [inline pom.xml]: Tabs/spaces

**Comment by eelhazati** [comment]
Tabs removed

**Pull request merged**